### ISSUE TITLE
Stash list format?

### ISSUE THREAD

**Issue by zenspider** [magit/magit#2846]
I'd like to have the ability to either customize the stash list format or, at the least, toggle on/off displaying the date of each stash.

Looking at the code in `magit-insert-stashes` All I want to do is to be able to toggle between:

```
git reflog --date=short --pretty=format:"%gd %at %gs" refs/stash
```

and the current:

```
git reflog  --pretty=format:"%gd %at %gs" refs/stash
```

**Comment by kyleam** [comment]
Doesn't the `--date=short` format often lead to revisions that are
ambiguous?  For example, I just created two test stashes, and this
gives me

```
$ git reflog --date=short --pretty=format:"%gd %at %gs" refs/stash
stash@{2016-10-31} 1477962385 On master: 63e2c3a AUTHORS.md: update list of contributors
stash@{2016-10-31} 1477962377 On master: 63e2c3a AUTHORS.md: update list of contributors
```

`magit-log-wash-rev` sets the first value as the identifier (used,
e.g., to show the Magit stash buffer), so it needs to be unique.

**Comment by zenspider** [comment]
That's a good point. I hadn't hit that yet... I'd be fine being able to customize the format string as well. I just want to be able to easily tell how old some of my stashes are.

**Comment by tarsius** [comment]
<kbd>L t</kbd> to show the "log margin" which contains the age. For stashes that only appears to  work in the status buffer. In the buffer `magit-stash-list` it doesn't, but that's a bug which probably isn't too hard to fix.

**Comment by kyleam** [comment]
> but that's a bug which probably isn't too hard to fix.

Hopefully it's fixed by #2847.

**Comment by tarsius** [comment]
Thanks!
